qt/place_panel.cpp

```diff
@@ -58,12 +57,68 @@ class QHLine : public QFrame
   }
 };
 
-PlacePageDialogUser::PlacePageDialogUser(QWidget * parent, place_page::Info const & info,
-                                         search::ReverseGeocoder::Address const & address)
-  : QDialog(parent)
+void PlacePanel::addCommonButtons(QDialogButtonBox * dbb, bool shouldShowEditPlace)
+{
+  dbb->setCenterButtons(true);
+
+  QPushButton * fromButton = new QPushButton("From");
+  fromButton->setIcon(QIcon(":/navig64/point-start.png"));
+  fromButton->setAutoDefault(false);
+  connect(fromButton, &QAbstractButton::clicked, this, [this](){ routeFrom(*infoPtr); });
+  dbb->addButton(fromButton, QDialogButtonBox::ActionRole);
+
+  QPushButton * addStopButton = new QPushButton("Stop");
+  addStopButton->setIcon(QIcon(":/navig64/point-intermediate.png"));
+  addStopButton->setAutoDefault(false);
+  connect(addStopButton, &QAbstractButton::clicked, this, [this](){ addStop(*infoPtr); });
+  dbb->addButton(addStopButton, QDialogButtonBox::ActionRole);
+
+  QPushButton * routeToButton = new QPushButton("To");
+  routeToButton->setIcon(QIcon(":/navig64/point-finish.png"));
+  routeToButton->setAutoDefault(false);
+  connect(routeToButton, &QAbstractButton::clicked, this, [this](){ routeTo(*infoPtr); });
+  dbb->addButton(routeToButton, QDialogButtonBox::ActionRole);
+
+  if (shouldShowEditPlace)
+  {
+    QPushButton * editButton = new QPushButton("Edit Place");
+    connect(editButton, &QAbstractButton::clicked, this, [this](){ editPlace(*infoPtr); });
+    dbb->addButton(editButton, QDialogButtonBox::AcceptRole);
+  }
+}
+
+PlacePanel::PlacePanel(QWidget * parent)
+  : QWidget(parent)
+{
+  setMaximumWidth(kMaximumPanelWidth);
+  setMinimumWidth(kMinimumPanelWidth);
+}
+
+void PlacePanel::setPlace(
+  place_page::Info const & info,
+  search::ReverseGeocoder::Address const & address
+)
+{
+  infoPtr = &info;
+
+  bool developerMode;
+  if (settings::Get(settings::kDeveloperMode, developerMode) && developerMode)
+    updateInterfaceDeveloper(info, address);
+  else
+    updateInterfaceUser(info, address);
+}
+
+void PlacePanel::updateInterfaceUser(
+  place_page::Info const & info,
+  search::ReverseGeocoder::Address const & address
+)
 {
   auto const & title = info.GetTitle();
 
+  if(this->layout() != nullptr){
+    QWidget().setLayout(this->layout());
+  }
```

Comment:
Something strange here. Did you mean?
```
if (auto const pLayout = layout())
  QWidget::setLayout(pLayout);
```

Otherwise QWidget().setLayout looks like setting something for the temporary object.

Reply:
Please, take it out as a separate function DeleteExistingLayout with a comment to SOF link.